logzio/datasource_metrics_account.go

```diff
@@ -0,0 +1,117 @@
+package logzio
+
+import (
+	"context"
+	"fmt"
+	"github.com/avast/retry-go"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/logzio/logzio_terraform_client/metrics_accounts"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func dataSourceMetricsAccount() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceMetricsAccountReadWrapper,
+		Schema: map[string]*schema.Schema{
+			metricsAccountId: {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			metricsAccountName: {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			metricsAccountEmail: {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			metricsAccountToken: {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			metricsAccountPlanUts: {
+				Type:     schema.TypeInt,
+				Optional: true,
+			},
+			metricsAccountAuthorizedAccounts: {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeInt,
+				},
+				Optional: true,
+			},
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Second),
+		},
+	}
+}
+
+func dataSourceMetricsAccountReadWrapper(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var err error
+	readErr := retry.Do(
+		func() error {
+			// we use this instead of return dataSourceMetricsAccountRead(d, m) directly
+			// to avoid 'no "id" found in attributes' error
+			if err = dataSourceMetricsAccountRead(d, m); err != nil {
```

Comment:
you're checking `err != nil` then returning `err` or `nil` based on that. This is redundant as you can directly return `err` :)

Reply:
when returning it directly it's causing an error from terraform `no "id" found in attributes` for some reason:
``` go
func() error {
	return dataSourceMetricsAccountRead(d, m)
}
```
and when using the syntax as used in [sub accounts](https://github.com/logzio/terraform-provider-logzio/blob/master/logzio/datasource_subaccount.go#L88-L95) it's causing a warning from cycode-security like in #163